aspnetcore/grpc/performance.md

```diff
@@ -215,3 +215,44 @@ payload.Data = UnsafeByteOperations.UnsafeWrap(data);
 Bytes are not copied with `UnsafeByteOperations.UnsafeWrap` so they must not be modified while the `ByteString` is in use.
 
 `UnsafeByteOperations.UnsafeWrap` requires [Google.Protobuf](https://www.nuget.org/packages/Google.Protobuf/) version 3.15.0 or later.
+
+## Read binary payloads
+
+Data can be efficiently read from `ByteString` instances by using `ByteString.Memory` and `ByteString.Span` properties.
+
+```csharp
+var byteString = UnsafeByteOperations.UnsafeWrap(new byte[] { 0, 1, 2 });
+var data = byteString.Span;
+
+for (var i = 0; i < data.Length; i++)
+{
+    Console.WriteLine(data[i]);
+}
+```
+
+These properties allow an app to read data directly from a `ByteString` without allocations or copies.
+
+Most .NET APIs have `ReadOnlyMemory<byte>` and `byte[]` overloads, so `ByteString.Memory` can normally be used. However, there are circumstances where an app might need to get the data as a byte array. If a byte array is required then the `MemoryMarshal.TryGetArray<T>` method can be used to get an array from the `ByteString` without allocating a new copy of the data.
+
+```csharp
+var byteString = GetByteString();
+
+ByteArrayContent content;
+if (MemoryMarshal.TryGetArray<byte>(byteString.Memory, out var segment))
+{
+    content = new ByteArrayContent(segment.Array, segment.Offset, segment.Count);
+}
+else
+{
+    content = new ByteArrayContent(byteString.ToByteArray());
```

Comment:
Add a comment here, saying that ToByteArray will make a copy of the data because TryGetArray failed.

Reply:
The explaination below covers this. Comments in documentation code are discouraged because they aren't translated.

Reply:
>  Comments in documentation code are discouraged because they aren't translated.

Code comments are fine and encouraged. Adding them to the text so they are translated is a major bonus. [We done a survey](https://github.com/MicrosoftDocs/feedback/issues/2515) and found out surprisingly, non-english speakers don't mind code comments, but they appreciate them in the translated portion of the doc. Having  them in both places is the best of everything. Often the MT (machine translation) is imperfect but they can figure it out from the english code comments.

Reply:
ok!